ROS abstraction layer

brendanburkhart opened the pull request:
Adds `crtk.ral` to provide a single common interface to both rospy and rclpy, so that other CRTK code can be agnostic to the ROS version.

`crtk.utils`, `crtk.wait_handle`, etc. have been updated to use `crtk.ral`, and so were the CRTK example scripts.

brendanburkhart committed:
Overhaul ros_12 to abstract away ros version

brendanburkhart committed:
Rename ros_12 to ral

ral: ROS abstraction layer

brendanburkhart committed:
Fix ros_12->ral in package __init__.py

brendanburkhart committed:
check_connections() now also checks all children

recursively checks children by default, can be disabled

brendanburkhart committed:
Add all twist/wrench message conversions

brendanburkhart committed:
Rewrite CRTK examples for new ral

brendanburkhart committed:
Mark crtk_servo_cv_example.py as executable

brendanburkhart committed:
Add create_rate() to ral

brendanburkhart committed:
Clean up RAL interface

brendanburkhart committed:
Use body/servo_cf in crtk_haptic_example

brendanburkhart committed:
Add script files to crtk python package

brendanburkhart committed:
Fix format string

brendanburkhart committed:
Port crtk.joystick_button to ROS 2

brendanburkhart committed:
Fix crtk_python_client setup.py

brendanburkhart committed:
Use ral.check_connections() in CRTK examples

brendanburkhart committed:
Revert

brendanburkhart committed:
Merge branch 'ros_abstraction_layer' of https://github.com/collaborative-robotics/ros2_crtk_python_client into ros_abstraction_layer

brendanburkhart committed:
Fix typo

adeguet1 reviewed (approved):
Really nice overall.  I have a few minor comments:
- [ ] Can you lower the d gain in the haptic example from 30 to 10, I believe some devices (Falcon maybe) were unstable with a higher gain
- [ ] Can you restore the amplitude/size of cube in examples?  Again, on some devices, it was safer to use a small displacement.  Also, the comments were correct, when we set amplitude to 1.0 and the comment says 2.0 since the amplitude is more like a radius
- [ ] In servo_cv example, I think code can be greatly simplified.  I would remove the loo on self.samples and just send a single servo_cv at a very slow speed (say 1mm per second), sleep for 10 seconds so the arm moves a cm and then call hold or a move_cv(0).  The issue here is that we don't have a system to test this example since the dVRK doesn't support servo_cv
- [ ] For all examples, can you update them to use the python argparse.

brendanburkhart committed:
Use argparse, correct haptic example

- Use argparse pkg in all CRTK examples
- Revert dgain/cube size changes in crtk_haptic_example.py
